lib/IRGen/GenCall.cpp

```diff
@@ -3246,14 +3268,18 @@ static void externalizeArguments(IRGenFunction &IGF, const Callee &callee,
                 == SILFunctionTypeRepresentation::Block) {
     // Ignore the physical block-object parameter.
     firstParam += 1;
-    // Or the indirect result parameter.
+  } else if (callee.getRepresentation() ==
+             SILFunctionTypeRepresentation::CXXMethod) {
+    // Skip the "self" param.
+    paramEnd--;
+    params = params.drop_front();
```

Comment:
This reads in a very confusing manner because dropping from the front makes it look like the start should be updated and not the end. Is this logically equivalent to the `firstParam += 1` handling in the block case? That doesn't modify `params`, so I'm not sure why we are modifying `params` here.

Reply:
Heh, this is why we need some more C++ method tests. The `params.drop_front()` part isn't supposed to be there. Remember, the self parameter is now last. We could drop the last param, but that doesn't seem necessary because we're going to have to update `paramEnd` anyway. 

Maybe `paramCount` would be a better name, though.